src/Rule/CharacterClassRule.php

```diff
@@ -21,7 +21,7 @@ abstract class CharacterClassRule implements Rule
     /**
      * @var CountConstraint[] Rule constraints.
      */
-    private $constraints;
+    private $constraints = [];
```

Comment:
While the `$this->constraints[] = ...` call makes it implicit, good to be explicit I think.

Reply:
Agreed :)